### PR TITLE
add softmax function tensorflow frontend

### PR DESCRIPTION
**Pull request by zuruoke**

### CODE REVIEW

**Comment by zuruoke** [comment]
https://github.com/unifyai/ivy/issues/4819

File: ivy/functional/frontends/tensorflow/raw_ops.py [hunk 1]
```diff
@@ -131,6 +131,12 @@ def Square(*, x, name="Square"):
     return ivy.square(x)
 
 
+def Softmax(*, x, axis=None, name="Softmax"):
+    if x.dtype not in ["half", "float32", "float64"]:
+        x = ivy.astype(x, "float64")
```
Comment (jkeane508) [inline ivy/functional/frontends/tensorflow/raw_ops.py:136]: I think that we should cast to float, but not necessarily float64 in all cases. We should maybe look at casting to the nearest reasonable float dtype instead

**Review by jkeane508 — CHANGES_REQUESTED** [review]

**Review by jkeane508 — CHANGES_REQUESTED** [review]
The test for Softmax is also failing [here](https://github.com/unifyai/ivy/actions/runs/3113328790/jobs/5047820850)

**Comment by jkeane508** [comment]
Feel free to request a review form me whenever you have tests passing :)

**Review by jkeane508 — CHANGES_REQUESTED** [review]
It seems to be from the testing that there are failures when the dims argument and the number of dims in the input for the function don't line up. I would suggest looking at the mathematics for softmax and creating a helper function that doesn't generate invalid data :)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
